ConnectionManager.php

```diff
@@ -0,0 +1,190 @@
+<?php
+include_once("iDatabase.php");
+/**
+ * Created by PhpStorm.
+ * User: gkalligeros
+ * Date: 11/18/2017
+ * Time: 3:14 PM
+ * Main class connection manager depends upon an database abstraction and a CacheManager
+ *
+ */
+class ConnectionManager
+{
+
+
+    private $db, $cacheManager;
+
+    /**
+     * ConnectionManager constructor.
+     * @param iDataBase $db
+     * @param CacheManager $cacheManager
+     */
+    public function __construct(iDataBase $db, CacheManager $cacheManager)
+    {
+        $this->db = $db;
+        $this->db->connect();
+        $this->cacheManager = $cacheManager;
+    }
+
+    /**
+     * Get requested value from cache if it is available or
+     * Generate a select statement with bindings and pass it to database for execution  then get the result store it in cache and return it
+     * @param string $tableName
+     * @param array $selectColumns
+     * @param  array $conditions
+     * @return bool
+     */
+    public function select($tableName, $selectColumns, $conditions)
+    {
+        $parsedColumns = implode(',', $selectColumns);
+        $query = "SELECT {$parsedColumns}  FROM {$tableName} WHERE ";
+        $conditionsParsed = [];
+        $bindings = [];
+        $result = $this->getCachedQueryResult($tableName, $selectColumns, $conditions);
+        if ($result != null) {
+            return $result;
+        }
+        foreach ($conditions as $condition) {
+            $placeHolderName = uniqid();
+            $conditionsParsed[] = " {$condition['column']} {$condition['operator']} :{$placeHolderName}";
+            $bindings[$placeHolderName] = $condition['value'];
+        }
+        $query .= implode(' AND ', $conditionsParsed);
+        $result = $this->db->execute($query, $bindings);
+        $this->cacheManager->remember($tableName, $selectColumns, $conditions, $result);
+        return $result;
+    }
```

Comment:
@tsirigotis @gkalligeros Why have you chosen not to provide the CRUD operations through the `iDatabase` interface?

Reply:
You mean the CRUD in the cache or in the database?

Reply:
@tsirigotis I mean the methods for selecting, inserting, deleting or updating records, Why have chosen to fully implement them inside the `ConnectionManager` and not in the `DummyDatabase` class. I would expect that those method would be part of the `iDatabase` interface

Reply:
I wanted to keep database driver as simple as possible  just one execute method.On a second thought your solution would had made my work much simpler.